core/wiki/macros/tree.tid

```diff
@@ -6,28 +6,31 @@ tags: $:/tags/Macro
 \end
 
 \define leaf-node(prefix,chunk)
+\whitespace trim
 <li>
 <$list filter="[<__prefix__>addsuffix<__chunk__>is[shadow]] [<__prefix__>addsuffix<__chunk__>is[tiddler]]" variable="full-title">
 <$list filter="[<full-title>removeprefix<__prefix__>]" variable="chunk">
-<span>{{$:/core/images/file}}</span> <$macrocall $name="leaf-link" full-title=<<full-title>> chunk=<<chunk>>/>
+<span>{{$:/core/images/file}}</span>&#32;<$macrocall $name="leaf-link" full-title=<<full-title>> chunk=<<chunk>>/>
```

Comment:
There are some `tc-xxx-gap` classes, that will allow you to add CSS gaps left, right and center if needed. So "artificial" spaces are not needed. 

This gap here can be added to `<$link to=<<__full-title__>>><$text text=<<__chunk__>>/></$link>` in the `leaf-link` macro from line 4. Like so: `<$link to=<<__full-title__>> class="tc-tiny-gap-left"><$text text=<<__chunk__>>/></$link>`

The other elements may need an additional span for proper "spacing".

Reply:
Is that the actual design preference to use those classes rather than `&#32;`? Or, as I saw used all over the place, `&nbsp;`? My aim when creating these PRs was to reduce changes to the resulting parse tree as much as possible, so a few deliberately placed spaces was better for that than introducing classes.

Is there a final decision on this? [I was following Jeremolene's remarks from here](https://github.com/Jermolene/TiddlyWiki5/pull/6241#issuecomment-973120667), but I don't know if he remembered the classes at the time either. I hadn't known about them.

Reply:
The classes are not very old. They have been created, because whitespace shouldn't be used for styling. CSS should be used for styling. ... 

IMO styled SPANs are semantically correct HTML code. Whitespace is _not_.

Reply:
OK. You are right.

Reply:
I'm not convinced about the new spacing classes. It seems to me that using whitespace for certain purposes like separating controls from their text is perfectly reasonable because a unit of whitespace has a clear semantic meaning.

Using `&#32;` is clever, but existing code tends to use `<$text text=" "/>`. I've no objection to a blanket change, but again it would require care. In the meantime, we should try to be consistent, or at least not make the inconsistencies worse.

Reply:
@Jermolene, So no worries about `<$text text=" "/>`. They have all already been removed as a part of #5473. The only one that remains is on inside the "menubar" plugin, and [it is literally having no effect](https://github.com/Jermolene/TiddlyWiki5/blob/4f42df8befd9afe9ff7647f027cd10355bb7f0a2/plugins/tiddlywiki/menubar/menu.tid#L17) (It's not being used with \whitespace trim).

@pmario, I hadn't considered the copy-paste implications of those spaces. That does seem an argument for removing them in favor of spacing classes, but I still of the opinion that it should be a separate issue from the "\whitespace trim" additions.

Reply:
@flibbles You are right. I'm OK with `&#32;` here. I don't think, there is a copy paste side effect here.